packages/augur-sdk/src/state/db/AbstractDB.ts

```diff
@@ -43,15 +46,17 @@ export abstract class AbstractDB {
     }
   }
 
-  protected async upsertDocument(id: string, document: object): Promise<PouchDB.Core.Response> {
+  protected async upsertDocument(id: string, document: object): Promise<PouchDB.UpsertResponse> {
```

Comment:
I don't *love* this implementation, as it will overrwite things if the document that is passed in has fields that are changed by another writer.

The reason it takes a thunk is so that when retrying you get the current version of the document to modify, so that your logic for determining how to do the update is fully run against the current version.

Now the basic version will probably work for us in many cases, but since I haven't seen all the call sites I would prefer that this not be the default behavior, and instead it should take a thunk like `upsert`.

Reply:
@pgebheim Turns out this conflict only happens with SyncStatus updates. I think the reason is that for all other things, it's an insert. SyncStatus is the only time we're modifying a record.

So...while I understand your concern here....IMNSHO it falls into the 'done not better' category. 

Obviously up for debate @justinbarry @pgebheim. I'm happy to modify this myself and make it pull a new doc and merge the two.